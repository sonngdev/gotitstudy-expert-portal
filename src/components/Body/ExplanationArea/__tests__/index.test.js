import React from 'react';
import { shallow } from 'enzyme';
import { ExplanationArea } from '../index';

describe('<ExplanationArea />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ExplanationArea />);
  });

  it('renders', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('contains a title and a subtitle', () => {
    expect(wrapper.text()).toContain('Problem explanation');
    expect(wrapper.text()).toContain('Complete all 3 sections before submitting your explanation');
  });

  it('contains three sections, each with title and editor', () => {
    expect(wrapper.find('h5')).toHaveLength(3);
    expect(wrapper.find('TinyEditor')).toHaveLength(3);
  });

  it('has Theory or Concept as the first section', () => {
    expect(wrapper.find('h5').at(0).text()).toBe('Theory or Concept');
  });

  it('has Step By Step Instructions as the second section', () => {
    expect(wrapper.find('h5').at(1).text()).toBe('Step By Step Instructions');
  });

  it('has Final Answer as the third section', () => {
    expect(wrapper.find('h5').at(2).text()).toBe('Final Answer');
  });
});
