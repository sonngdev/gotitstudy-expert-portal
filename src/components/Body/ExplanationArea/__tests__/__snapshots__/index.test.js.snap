// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExplanationArea /> renders 1`] = `
"<div className=\\"explanation-area\\">
  <h3 className=\\"u-textCapitalize u-marginBottomTiny\\">
    Problem explanation
  </h3>
  <p className=\\"u-opacityHalf\\">
    Complete all 3 sections before submitting your explanation
  </p>
  <div className=\\"explanation-editors u-paddingRightSmall\\">
    <h5>
      Theory or Concept
    </h5>
    <TinyEditor id=\\"concept-editor\\" value=\\"\\" onChange={[Function: onChange]} config={{...}} />
    <h5 className=\\"u-marginTopLarge\\">
      Step By Step Instructions
    </h5>
    <TinyEditor id=\\"instructions-editor\\" value=\\"\\" onChange={[Function: onChange]} config={{...}} />
    <h5 className=\\"u-marginTopLarge\\">
      Final Answer
    </h5>
    <TinyEditor id=\\"answer-editor\\" value=\\"\\" onChange={[Function: onChange]} config={{...}} />
  </div>
  <ExplanationFooter explanations={{...}} />
</div>"
`;
